### DISCUSSION TITLE
Answering Machine Detection

### DISCUSSION THREAD

**Discussion by iamadamreed** [rtckit/eqivo#4] [Q&A] [answered]
Hello, I was curious if you've pursued implementing AMD with this framework?

**Comment by cdosoftei** [comment]
Hi @adamreed90!

At this point in time the API should be on par with [plivo/plivoframework](https://github.com/plivo/plivoframework) which never implemented AMD as a first class feature. It doesn't mean it cannot be implemented as a backwards compatible add-on!

I've successfully implemented AMD scenarios, however unbeknownst to Eqivo: [webitel/mod_amd](https://github.com/webitel/mod_amd) is a rather advanced fork of the original module so with it bundled in the FreeSWITCH image, it can be invoked when [originating calls](https://eqivo.org/#v0-1-call), specifically the `ExtraDialString` parameter should include `execute_on_answer=amd`.

As a first class feature, this wouldn't be necessary and webhooks are also possible as the recommended `mod_amd` fork fires FreeSWITCH events once the analysis is complete.

**Comment by cdosoftei** [comment, marked as answer]
Hi @adamreed90 --

Answering machine detection is now (as of [v0.5.4](https://github.com/rtckit/eqivo/releases/tag/v0.5.4) :tada:) implemented as a first calls feature, this is achieved in a [Twilioesque](https://www.twilio.com/docs/voice/answering-machine-detection) fashion, by passing the `MachineDetection` parameter when [placing new calls](https://eqivo.org/#v0-1-call).

The detection itself is performed via [mod_amd](https://github.com/rtckit/mod_amd) and mod_avmd (shipped with FreeSWITCH). A compatible Docker image is readily available: [rtckit/eqivo-freeswitch](https://hub.docker.com/r/rtckit/eqivo-freeswitch).

**Reply by iamadamreed** [reply]
Awesome! Great work!

**Comment by iamadamreed** [comment]
Is it possible to do this in an asynchronous method, like Twilio does for example?

**Reply by cdosoftei** [reply]
Yes! Check out the `AsyncAMD` parameter here: https://eqivo.org/#v0-1-call

**Reply by iamadamreed** [reply]
I'm sorry I missed that, thank you so much for your help, can't wait to give this a go!
